### ISSUE TITLE
rewrite steve to use click

### ISSUE THREAD

**Issue by willkg** [pyvideo/steve#32]
click is a library for building command line utilities. I've started switching all my utilities to use it since it makes writing clis much easier.

http://click.pocoo.org/5/

We should switch steve to use click. It'll let us drop a bunch of code. It'll improve the testability of steve. It'll make it easier to do some things we can't do now.

**Comment by AvdN** [comment]
click is internally build upon the outdated optparse. That is not a problem in itself, but if you want to do something under the hood, you'll find that familiarity with the more modern argparse and its concepts won't help you. E.g. try to find from the documentation if you can have  subcommand aliases; or can (re-)use special help formatters; or do simple things like specifying what optional argument is to be stored as a truth value (it is not the ``action="store_true"`` developers are already familiar with)

For me that, and other things, were the reason not to use `click` and to create [ruamel.std.argparse](https://bitbucket.org/ruamel/std.argparse#rst-header-wrapping-argparse) which wraps around, and enhances the standard argparse module, but provides a more familiar interface e.g. for its `@option` decorator than `click` does.

 It operates closely with [ruamel.appconfig](https://pypi.python.org/pypi/ruamel.appconfig) to allow loading (and storing) of default values for your commandline arguments in `~/.config/steve/steve.yaml` (or `steve.ini` if you prefer that format over YAML)

If you are open to that as an alternative to `click` let me know and I will look at updating `cmdline.py` and provide a  PR.

**Comment by willkg** [comment]
`ruamel.std.argparse` looks interesting as does the entire ruamel suite.

I have a bunch of experience with click already that I've accrued over the last year. I got pretty sick of the optparse/argparse situation and looked for better APIs for clis and ended up evaluating four or five of which I only remember click, clint and cliff. click was the one I liked the most; the docs are decent, the API was easy to wrap my head around and it did everything I wanted including dealing with pipes and some other things beyond just argument parsing. The only issue I had was with the way help text works, but I worked around that with dennis (http://dennis.readthedocs.org/en/latest/) in a good enough manner that it wasn't a deal breaker.

I think I want to continue with click but if steve has requirements that click can't handle well, then I'll look at `ruamel.std.argparse` again.

**Comment by AvdN** [comment]
OK, it makes sense to go with what you already know to work. I just wanted to make sure that before I did invest the time to provide a PR for this , to check that there is a chance of acceptance. Now I can do something else :-)

I will look at steve at some time anyway (otherwise I would not have bothered to watch the project), looking how what I did for uploading/generating EuroPython Conference video info can contribute to the project.
